packages/flutter/lib/src/material/banner.dart

```diff
@@ -318,6 +326,7 @@ class _MaterialBannerState extends State<MaterialBanner> {
     );
 
     final double elevation = widget.elevation ?? bannerTheme.elevation ?? 0.0;
+    final EdgeInsetsGeometry margin = widget.margin ?? EdgeInsets.only(bottom: elevation > 0 ? 10.0 : 0.0);
```

Comment:
I think it would be more clear to use `EdgeInsets.only(bottom: elevation > 0 ? 10.0 : 0.0)` as the default value for `widget.margin` in the constructor, and to document it as such.  That way users know what they're getting if they don't set this, or if they want to change it slightly.

Reply:
Hi @justinmc Thanks for your valuable input. Shall I go ahead and mak the changes?

Reply:
Yes please!

Reply:
We make the constructor parameters null by default to allow themes to specify an override for the default value. Although there's no MaterialBannerTheme now, if we added one in the future, this default parameter would get in the way. For now having the constructor parameters be null by default is still useful, since it makes them consistent with the rest of the Material library's components.

Reply:
My mistake, that's right!  I think this is fine as-is then.

Reply:
Having all constructors consistent in Material Library's components makes a lot of sense.  It might help a lot in future adaptability as @HansMuller mentioned.
@justinmc I  will keep the implementation the same as before and not make the changes for now.
@HansMuller Let me know if I should make the changes from the null value in the constructor to the default value of `margin` to `EdgeInsets.only(bottom: elevation > 0 ? 10.0: 0.0)`